pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -614,6 +637,13 @@ func (gce *GCECloud) targetPoolURL(name, region string) string {
 	return fmt.Sprintf("https://www.googleapis.com/compute/v1/projects/%s/regions/%s/targetPools/%s", gce.projectID, region, name)
 }
 
+func (gce *GCECloud) GetLoadBalancerName(service *v1.Service) string {
+	//GCE requires that the name of a load balancer starts with a lower case letter.
+	ret := "a" + service.Namespace[0:13] + "-" + service.Name[0:13] + "-" + string(service.UID)
+	ret = strings.Replace(ret, "-", "", -1)
```

Comment:
above you build a string like "namespace-service-abcd-12-34-5678" and then here you remove all the dashes - why?